### ISSUE TITLE
[Android] layout: Whole screen is pushed, when keyboard pops.

### ISSUE THREAD

**Issue by jainkuniya** [zulip/zulip-mobile#3010]
The app bar is gone; plus the input is partially obscured by the keyboard.

Tested on Pixel 2 emulator with Android 8 and One plus six with Android 9.0

![](https://user-images.githubusercontent.com/18511177/45839321-ad19da80-bd31-11e8-830a-c21aaa285564.png)

![](https://user-images.githubusercontent.com/28173/45979654-9e1e7b00-c004-11e8-8694-14b06604ce50.png)

**Comment by jainkuniya** [comment]
This is completely wried.
Steps to reproduce:
* re-install app on Android (run `react-native run-android`)
* This bug can be reproduced.

Now force close the app.
* reopen app.
* this bug can't be reproduced.


https://drive.google.com/file/d/1TUyaN04PZMWWKQ-d-I7aktFex7yUnQBC/view?usp=sharing

**Comment by borisyankov** [comment]
I have definitely seen that in an Android emulator, but not on my machine.

Worth trying to reproduce in a minimal app and reporting to React Native as it sounds like a bug in the `KeyboardAvoidingView` component.

**Comment by jainkuniya** [comment]
But for android we aren't using RN `KeyboardAvoidingView`, we are just returning `View`

```
 if (Platform.OS === 'android') {
      return <View style={style}>{children}</View>;
    }
```

**Comment by borisyankov** [comment]
Yeah, of course :)
I did some more research, even with the `KeyboardAvoidingView` it still acts the same.
Changing the keyboard behavior in the `AndroidManifest.xml` also did not fix/affect this issue.

**Comment by gnprice** [comment]
Hmm interesting.

I saw some behavior like this with #2919 , described here: https://github.com/zulip/zulip-mobile/pull/2919#issuecomment-424131638 . (And @jainkuniya , it looks like your second screenshot is taken from that?) But I've never seen it on any version in master.

Does the repro in the video work reliably?

**Comment by jainkuniya** [comment]
> Does the repro in the video work reliably?

Yup! I can reproduce this issue on the first launch of the app.

**Comment by borisyankov** [comment]
To summarize our chat...
I am working a fix but before implementing it I'd like to know what devices are affected.
I also did reproduce on an emulator but not on a real device.

**Comment by borisyankov** [comment]
OK, confirmed that this is related to an updated `Yoga` version that is coming with RN 0.56 / 0.57
So it is a blocker for the upgrade, but luckily I did found a solution yesterday 🥇

**Comment by gnprice** [comment]
To make sure this point is explicit:
* This issue seems to happen 100% of the time on the first run of the app after install.
* It's not just once -- it happens again and again, as many times as the user tries, within that first run.

That (a) is a fascinating pattern that must be a clue somehow; and (b) means that it will matter quite a bit for a user's first impression of the app. If they see it a couple of times when they first try the app, it's likely to stick in their mind as an indication that the app is buggy and unreliable; they'll have no reason to immediately conclude it's gone when they stop seeing it.

**Comment by borisyankov** [comment]
But does that happen on real devices without Debugging enabled? 
My suspicion is that it might be related to the logging and some delays.
We should try that with a newly-built APK on a phone to verify.

**Comment by gnprice** [comment]
Yeah, agree -- that'll be a good thing to check. If we can convince ourselves it only happens in debug mode, not release, then that's a different way of resolving it off the priority list :slightly_smiling_face:
